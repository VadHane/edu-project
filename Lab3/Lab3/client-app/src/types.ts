export type Nullable<T> = T | null;

export type Maybe<T> = T | undefined;

export type IRoute = {
    path: string;
    element: JSX.Element;
    exact?: boolean;
};

export enum RouteNamesEnum {
    Users = '/users',
    AddUser = '/users/add',
    EditUserById = '/users/edit/:id',
}
