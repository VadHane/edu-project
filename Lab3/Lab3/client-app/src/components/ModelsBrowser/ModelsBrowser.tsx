import {
    TableContainer,
    Paper,
    Table,
    TableHead,
    TableRow,
    TableCell,
    TableBody,
    styled,
    tableCellClasses,
    CircularProgress,
} from '@mui/material';
import React, { FunctionComponent, useEffect, useState } from 'react';
import { Model } from '../../models/Model';
import BrowsersRow from './BrowsersRow/BrowsersRow';
import {
    MODEL_NAME_LABEL,
    DESCRIPTION_LABEL,
    TAGS_LABEL,
    ACTIONS_LABEL,
    tableHeadBackgroundColor,
    tableHeadTextColor,
} from './ModelsBrowser.constants';
import { paperStyles, tableContainerStyles, tableStyles } from './ModelsBrowser.styles';
import { ModelsBrowserProps } from './ModelsBrowser.types';

const ModelsBrowser: FunctionComponent<ModelsBrowserProps> = (
    props: ModelsBrowserProps,
) => {
    const [models, setModels] = useState<Array<Model>>([]);

    const [isLoading, setIsLoading] = useState<boolean>(true);

    useEffect(() => {
        setModels([...props.models]);
    }, [props.models]);

    useEffect(() => {
        setIsLoading(props.isLoading);
    }, [props.isLoading]);

    const StyledTableCell = styled(TableCell)(() => ({
        [`&.${tableCellClasses.head}`]: {
            backgroundColor: tableHeadBackgroundColor,
            color: tableHeadTextColor,
            textAlignLast: 'center',
        },
        [`&.${tableCellClasses.body}`]: {
            fontSize: 14,
        },
    }));

    const tablesBody: React.ReactNode = (
        <>
            {models.length > 0 ? (
                models.map((model) => <BrowsersRow key={model.id} model={model} />)
            ) : (
                <TableRow>
                    <TableCell align="center" colSpan={4}>
                        Not Found
                    </TableCell>
                </TableRow>
            )}
        </>
    );

    return (
        <Paper sx={paperStyles}>
            <TableContainer sx={tableContainerStyles}>
                <Table stickyHeader sx={tableStyles}>
                    <TableHead>
                        <TableRow>
                            <StyledTableCell>{MODEL_NAME_LABEL}</StyledTableCell>
                            <StyledTableCell>{DESCRIPTION_LABEL}</StyledTableCell>
                            <StyledTableCell>{TAGS_LABEL}</StyledTableCell>
                            <StyledTableCell>{ACTIONS_LABEL}</StyledTableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {isLoading ? (
                            <TableRow>
                                <TableCell align="center" colSpan={4}>
                                    <CircularProgress />
                                </TableCell>
                            </TableRow>
                        ) : (
                            tablesBody
                        )}
                    </TableBody>
                </Table>
            </TableContainer>
        </Paper>
    );
};

export default ModelsBrowser;
