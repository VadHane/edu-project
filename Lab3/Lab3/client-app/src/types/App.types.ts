export type Nullable<T> = T | null;

export type Maybe<T> = T | undefined;

export enum ModalResultActions {
    Add = 1,
    Edit = 2,
}
