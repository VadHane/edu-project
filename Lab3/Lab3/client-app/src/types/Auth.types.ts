import { User } from '../models/User';
import { Maybe } from '../types/App.types';

export type AuthState = {
    isAuth: boolean;
    user: Maybe<User>;

    isLoading: boolean;
    error: Nullable<string>;
};

export enum AuthActionTypes {
    AUTHORIZATION_START = 'AUTHORIZATION_START',
    AUTHORIZATION_SUCCESS = 'AUTHORIZATION_SUCCESS',
    AUTHORIZATION_ERROR = 'AUTHORIZATION_ERROR',

    REGISTRATION_START = 'REGISTRATION_START',
    REGISTRATION_SUCCESS = 'REGISTRATION_SUCCESS',
    REGISTRATION_ERROR = 'REGISTRATION_ERROR',

    LOGOUT_START = 'LOGOUT_START',
    LOGOUT_SUCCESS = 'LOGOUT_SUCCESS',
    LOGOUT_ERROR = 'LOGOUT_ERROR',
}

type AuthStartAction = {
    type: AuthActionTypes.AUTHORIZATION_START;
};

type AuthSuccessAction = {
    type: AuthActionTypes.AUTHORIZATION_SUCCESS;
    payload: User;
};

type AuthErrorAction = {
    type: AuthActionTypes.AUTHORIZATION_ERROR;
    payload: string;
};

type AuthActions = AuthStartAction | AuthSuccessAction | AuthErrorAction;

type RegStartAction = {
    type: AuthActionTypes.REGISTRATION_START;
};

type RegSuccessAction = {
    type: AuthActionTypes.REGISTRATION_SUCCESS;
    payload: User;
};

type RegErrorAction = {
    type: AuthActionTypes.REGISTRATION_ERROR;
    payload: string;
};

type RegActions = RegStartAction | RegSuccessAction | RegErrorAction;

type LogoutStartAction = {
    type: AuthActionTypes.LOGOUT_START;
};

type LogoutSuccessAction = {
    type: AuthActionTypes.LOGOUT_SUCCESS;
};

type LogoutErrorAction = {
    type: AuthActionTypes.LOGOUT_ERROR;
    payload: string;
};

type LogoutActions = LogoutStartAction | LogoutSuccessAction | LogoutErrorAction;

export type AuthorizationActions = AuthActions | RegActions | LogoutActions;

export enum FetchMethodsEnum {
    GET = 'GET',
    POST = 'POST',
    PUT = 'PUT',
    DELETE = 'DELETE',
}

export type FetchOptions = {
    method?: FetchMethodsEnum;
    body?: FormData;
};
