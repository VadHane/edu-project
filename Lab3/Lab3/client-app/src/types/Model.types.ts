import { Model } from '../models/Model';

export type ModelState = {
    models: Array<Model>;
};

export enum ModelActionTypes {
    GET_ALL_MODELS_SUCCESS = 'GET_ALL_MODELS_SUCCESS',

    ADD_MODEL_SUCCESS = 'ADD_MODEL_SUCCESS',

    EDIT_MODEL_SUCCESS = 'EDIT_MODEL_SUCCESS',

    DELETE_MODEL_SUCCESS = 'DELETE_MODEL_SUCCESS',
}

type GetModelsSuccessAction = {
    type: ModelActionTypes.GET_ALL_MODELS_SUCCESS;
    payload: Array<Model>;
};

type ModifyModelSuccessAction = {
    type:
        | ModelActionTypes.ADD_MODEL_SUCCESS
        | ModelActionTypes.EDIT_MODEL_SUCCESS
        | ModelActionTypes.DELETE_MODEL_SUCCESS;
    payload: Model;
};

export type ModelActions = GetModelsSuccessAction | ModifyModelSuccessAction;
