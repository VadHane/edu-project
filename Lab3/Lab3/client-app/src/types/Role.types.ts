import { Role } from '../models/Role';

export type RoleState = {
    roles: Array<Role>;
    loading: boolean;
    loaded: boolean;
    error: Nullable<string>;
};

export enum RolesActionTypes {
    GET_ALL = 'GET_ALL_ROLES',
    GET_ALL_SUCCESS = 'GET_ALL_ROLES_SUCCESS',
    GET_ALL_ERROR = 'GET_ALL_ROLES_ERROR',

    ADD_ROLE = 'ADD_ROLE',
    ADD_ROLE_SUCCESS = 'ADD_ROLE_SUCCESS',
    ADD_ROLE_ERROR = 'ADD_ROLE_ERROR',
}

type GetRolesStartAction = {
    type: RolesActionTypes.GET_ALL;
};

type GetRolesSuccessAction = {
    type: RolesActionTypes.GET_ALL_SUCCESS;
    payload: Array<Role>;
};

type GetRolesErrorAction = {
    type: RolesActionTypes.GET_ALL_ERROR;
    payload: string;
};

type GetRolesAction = GetRolesStartAction | GetRolesSuccessAction | GetRolesErrorAction;

type ModifyRolesStartAction = {
    type: RolesActionTypes.ADD_ROLE;
};

type ModifyRolesSuccessAction = {
    type: RolesActionTypes.ADD_ROLE_SUCCESS;
    payload: Role;
};

type ModifyRolesErrorAction = {
    type: RolesActionTypes.ADD_ROLE_ERROR;
    payload: string;
};

type ModifyRolesAction =
    | ModifyRolesStartAction
    | ModifyRolesSuccessAction
    | ModifyRolesErrorAction;

export type RoleAction = GetRolesAction | ModifyRolesAction;
