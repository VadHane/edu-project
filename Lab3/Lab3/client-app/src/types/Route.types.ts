export enum RouteNamesEnum {
    Users = '/users',
    Models = '/models',
    Add = '/add',
    EditById = '/edit/:id',

    Login = '/login',
    Registration = '/registration',
}
