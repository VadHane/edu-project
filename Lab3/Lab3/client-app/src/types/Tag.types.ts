import { Tag } from '../models/Tag';

export type TagState = {
    tags: Array<Tag>;
    loading: boolean;
    loaded: boolean;
    error: Nullable<string>;
};

export enum TagActionTypes {
    GET_ALL_TAGS = 'GET_ALL_TAGS',
    GET_ALL_TAGS_SUCCESS = 'GET_ALL_TAGS_SUCCESS',
    GET_ALL_TAGS_ERROR = 'GET_ALL_TAGS_ERROR',

    ADD_TAG = 'ADD_TAG',
    ADD_TAG_SUCCESS = 'ADD_TAG_SUCCESS',
    ADD_TAG_ERROR = 'ADD_TAG_ERROR',
}

type getTagsStartAction = {
    type: TagActionTypes.GET_ALL_TAGS;
};

type getTagsSuccessAction = {
    type: TagActionTypes.GET_ALL_TAGS_SUCCESS;
    payload: Array<Tag>;
};

type getTagsErrorAction = {
    type: TagActionTypes.GET_ALL_TAGS_ERROR;
    payload: string;
};

type getTagsAction = getTagsStartAction | getTagsSuccessAction | getTagsErrorAction;

type addTagStartAction = {
    type: TagActionTypes.ADD_TAG;
};

type addTagSuccessAction = {
    type: TagActionTypes.ADD_TAG_SUCCESS;
    payload: Tag;
};

type addTagErrorAction = {
    type: TagActionTypes.ADD_TAG_ERROR;
    payload: string;
};

type addTagActions = addTagStartAction | addTagSuccessAction | addTagErrorAction;

export type TagActions = getTagsAction | addTagActions;
