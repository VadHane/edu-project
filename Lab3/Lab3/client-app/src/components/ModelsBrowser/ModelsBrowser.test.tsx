import { render } from '@testing-library/react';
import React from 'react';
import ModelsBrowser from './ModelsBrowser';
import { ModelsBrowserProps } from './ModelsBrowser.types';

const props: ModelsBrowserProps = {
    models: [],
    isLoading: false,
};

jest.mock('./BrowsersRow', () => ({
    __esModule: true,
    default: () => <></>,
}));

describe('Test models browser.', () => {
    test('Snapshot', () => {
        const utils = render(<ModelsBrowser {...props} />);

        expect(utils).toMatchSnapshot();
    });
});
