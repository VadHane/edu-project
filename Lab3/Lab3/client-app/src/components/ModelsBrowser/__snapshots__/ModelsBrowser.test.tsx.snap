// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test models browser. Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-14ie8xk-MuiPaper-root"
      >
        <div
          class="MuiTableContainer-root css-1j7oosa-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root MuiTable-stickyHeader css-1razvkq-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                  scope="col"
                >
                  Model Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                  scope="col"
                >
                  Description
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                  scope="col"
                >
                  Tags
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                  scope="col"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
                  colspan="4"
                >
                  Not Found
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-14ie8xk-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-1j7oosa-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root MuiTable-stickyHeader css-1razvkq-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                scope="col"
              >
                Model Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                scope="col"
              >
                Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                scope="col"
              >
                Tags
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-1ck9330-MuiTableCell-root"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
                colspan="4"
              >
                Not Found
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
